extendr-api/src/robj/tests.rs

```diff
@@ -92,21 +94,87 @@ fn test_from_robj() {
     }
 }
 
-#[test]
-fn test_try_from_robj() {
+#[rstest]
+#[case(true)]
+#[case(false)]
+// NOTE: `u8` is missing
+#[case(1_u16)]
+#[case(1_u32)]
+#[case(1_u64)]
+#[case(1_usize)]
+#[case(1_i8)]
+#[case(1_i16)]
+#[case(1_i32)]
+#[case(1_i64)]
+// NOTE: `isize` is missing
+#[case(0_f32)]
+#[case(0_f64)]
+#[case(1_f32)]
+#[case(1_f64)]
+fn test_try_from_robj<T>(#[case] value: T)
+where
+    Robj: From<T>,
+    T: TryFrom<Robj>,
+    T: ToVectorValue,
+    T: Copy,
+    T: std::fmt::Debug + PartialEq,
+    <T as std::convert::TryFrom<robj::Robj>>::Error: std::fmt::Debug,
+    <T as std::convert::TryFrom<robj::Robj>>::Error: PartialEq,
+{
+    test! {
+        assert_eq!(TryFrom::try_from(Robj::from(value)), Ok(value));
+    }
+}
+
+#[rstest]
+// NOTE: `NonZeroU8` is missing
+#[case(std::num::NonZeroU16::new(1).unwrap())]
+#[case(std::num::NonZeroU32::new(1).unwrap())]
+#[case(std::num::NonZeroU64::new(1).unwrap())]
+#[case(std::num::NonZeroUsize::new(1).unwrap())]
+#[case(std::num::NonZeroI8::new(1).unwrap())]
+#[case(std::num::NonZeroI16::new(1).unwrap())]
+#[case(std::num::NonZeroI32::new(1).unwrap())]
+#[case(std::num::NonZeroI64::new(1).unwrap())]
+// NOTE: `NonZeroIsize` is missing
+fn test_try_from_robj_nonzero<Value>(#[case] value: Value)
+where
+    Robj: From<Value>,
+    Value: TryFrom<Robj>,
+    Value: Copy + std::fmt::Debug + PartialEq,
+    <Value as std::convert::TryFrom<robj::Robj>>::Error: std::fmt::Debug + PartialEq,
+{
+    test! {
+        assert_eq!(Value::try_from(dbg!(Robj::from(value))).ok(), Some(value));
+    }
+}
+
+#[rstest]
+#[case(std::num::NonZeroU8::new(1).unwrap())]
+#[case(std::num::NonZeroU8::new(1).unwrap())]
+#[case(std::num::NonZeroU16::new(1).unwrap())]
+#[case(std::num::NonZeroU32::new(1).unwrap())]
+#[case(std::num::NonZeroU64::new(1).unwrap())]
+#[case(std::num::NonZeroUsize::new(1).unwrap())]
+#[case(std::num::NonZeroI8::new(1).unwrap())]
+#[case(std::num::NonZeroI16::new(1).unwrap())]
+#[case(std::num::NonZeroI32::new(1).unwrap())]
+#[case(std::num::NonZeroI64::new(1).unwrap())]
```

Comment:
It's weird to not be using the values here, but I guess `rstest` doesn't support a turbofish invocation so maybe there is no better option.